### DISCUSSION TITLE
Need few codemod commands

### DISCUSSION THREAD

**Discussion by chidaquest** [webdriverio/codemod#70] [Migration Issue] [answered]
Hi presently I am using the usual codemod command: 

 npx jscodeshift -t .\node_modules\@wdio\codemod\protractor <file_path>

It rewrites the same file. I need codemod commands for the following: 
1. I need to keep the old file untouched and write the migration changes to a new file. 
2. Say I have a folder called .\e2e\steps under which there are multiple files. What is the codemod command to change all the files in the current folder. 
3. What is the command to add multiple files to the conversion. 
4. Is there a codemod command that can help to do migrate all files inside the parent directory including the files under its child folder.

**Comment by christian-bromann** [comment, marked as answer]
> I need to keep the old file untouched and write the migration changes to a new file.

Seems like JSCodeshift doesn't have that feature, feel free to create a feature request in that project

> Say I have a folder called .\e2e\steps under which there are multiple files. What is the codemod command to change all the files in the current folder.

I believe you can use the glob pattern to fetch a whole directory of files, e.g.

```sh
npx jscodeshift -t .\node_modules@wdio\codemod\protractor ./path/to/dir/**/*.e2e.js
```

> What is the command to add multiple files to the conversion.

What do you mean by that?

> Is there a codemod command that can help to do migrate all files inside the parent directory including the files under its child folder.

See answer in 2.

**Reply by chidaquest** [reply]
//What is the command to add multiple files to the conversion.

Say I have two files in two different places, for ex: 
1. e2e\pages\Login\login.po.ts
2. e2e\pages\HomePage\Mainmenu.po.ts

Can we convert  both these files in a simple command of codemod?

**Comment by christian-bromann** [comment]
Maybe:

```
npx jscodeshift -t .\node_modules@wdio\codemod\protractor ./path/to/dir/**/(login|Mainmenu).e2e.js
```

Not sure what the right glob pattern is.
